Lib/unittest/mock.py

```diff
@@ -2731,6 +2731,7 @@ def create_autospec(spec, spec_set=False, instance=False, _parent=None,
                               _new_parent=parent,
                               **kwargs)
             mock._mock_children[entry] = new
+            new.return_value = Mock()
```

Comment:
This could use the `child_klass` to construct the mock object instead of `Mock`. This can break code that depended on the return value being an instance of `child_klass`. Example case as below : 

```python
from unittest.mock import create_autospec, seal

class Foo:

    def bar(self):
        pass

spec = create_autospec(Foo)
print(spec.bar())
print(len(spec.bar()))
```

Python 3.10

```
python3.10 gh92213.py  
<MagicMock name='mock.bar()' id='140499591708000'>
0
```

With patch

```
./python gh92213.py
<Mock name='mock.bar()' id='140578691284304'>
Traceback (most recent call last):
  File "/home/karthikeyan/stuff/python/cpython/gh92213.py", line 10, in <module>
    print(len(spec.bar()))
          ^^^^^^^^^^^^^^^
TypeError: object of type 'Mock' has no len()
```

Reply:
That makes sense to me, I used Mock originally because I wasn't sure if MagicMock might "evade" `seal()` if set explicitly, but I tested it and it does not:

```
class Foo:
    x = 1
    def foo(self) -> int:
        return 0
foo=Foo()
foo = mock.create_autospec(foo)
foo.foo().x
```
==>  AttributeError: mock.foo().x

I've pushed the update to use `child_klass()`

Reply:
@akulakov - rather than just manually testing this, please can you add a test to the test suite to ensure it doesn't change in the future?

Reply:
@cjw296 that makes sense, I've expanded the unit test - thanks for noting this.